### PR TITLE
STORM-2225: change spout config to be simpler. (1.x)

### PR DESCRIPTION
**Pull request by revans2**

### CODE REVIEW

**Comment by revans2** [comment]
This is the 1.x version of #1808

**Comment by revans2** [comment]
The test failures are unrelated and are around the integration tests that always seem to fail lately.

File: examples/storm-kafka-client-examples/src/main/java/org/apache/storm/kafka/trident/TridentKafkaClientWordCountNamedTopics.java [hunk 1]
```diff
@@ -18,87 +18,58 @@
...
-                    kafkaSpoutStreams, newTuplesBuilder(), newRetryService())
+	private static Func<ConsumerRecord<String, String>, List<Object>> JUST_VALUE_FUNC = new Func<ConsumerRecord<String, String>, List<Object>>() {
+		@Override
+		public List<Object> apply(ConsumerRecord<String, String> record) {
+			return new Values(record.value());
+		}
+	};
+    
+    protected KafkaSpoutConfig<String,String> newKafkaSpoutConfig() {
+        return KafkaSpoutConfig.builder("127.0.0.1:9092", TOPIC_1, TOPIC_2)
```
Comment (ppoulosk) [inline examples/storm-kafka-client-examples/src/main/java/org/apache/storm/kafka/trident/TridentKafkaClientWordCountNamedTopics.java]: Why are we using a hardcoded port here?

Comment (revans2) [inline examples/storm-kafka-client-examples/src/main/java/org/apache/storm/kafka/trident/TridentKafkaClientWordCountNamedTopics.java]: This is an exact translation of the original code. Even down to not using KAFKA_LOCAL_BROKER.  If people want me to change it I am happy to, but I thought it best to not overreach on the scope of the pull request.  At least until the code worked.

File: external/storm-kafka-client/src/main/java/org/apache/storm/kafka/bolt/selector/FieldIndexTopicSelector.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    private final String defaultTopicName;
+
+    public FieldIndexTopicSelector(int fieldIndex, String defaultTopicName) {
+        this.fieldIndex = fieldIndex;
+        this.defaultTopicName = defaultTopicName;
+    }
+
+    @Override
+    public String getTopic(Tuple tuple) {
+        if (fieldIndex < tuple.size()) {
```
Comment (ppoulosk) [inline external/storm-kafka-client/src/main/java/org/apache/storm/kafka/bolt/selector/FieldIndexTopicSelector.java]: Protect against negative index?

File: external/storm-kafka-client/README.md [hunk 3]
```diff
@@ -1,192 +1,5 @@
-#Storm Kafka Spout with New Kafka Consumer API
+#Storm Kafka integration using the kafka-client jar
```
Comment (harshach) [inline external/storm-kafka-client/README.md]: can we call it as Kafka New consumer API. Not many understand its using newer api by calling it as "kafka-client" jar

File: external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/StormStringDeserializer.java [hunk 4]
```diff
@@ -15,22 +15,11 @@
...
-import java.util.List;
-
-public class KafkaSpoutTuplesBuilderWildcardTopics<K,V> implements KafkaSpoutTuplesBuilder<K,V> {
-    private KafkaSpoutTupleBuilder<K, V> tupleBuilder;
+import org.apache.kafka.common.serialization.StringDeserializer;
 
-    public KafkaSpoutTuplesBuilderWildcardTopics(KafkaSpoutTupleBuilder<K, V> tupleBuilder) {
-        this.tupleBuilder = tupleBuilder;
-    }
+public class StormStringDeserializer extends StringDeserializer implements SerializableDeserializer<String> {
```
Comment (harshach) [inline external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/StormStringDeserializer.java]: Do we still need this? users can configure the StringSerializer from kafka-clients lib

**Comment by harshach** [comment]
@revans2 overall looks good to me. Minor nits.

**Comment by ppoulosk** [comment]
+1, Non-binding.

File: external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/StormStringDeserializer.java (hunk 4 above)
Comment (revans2) [inline external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/StormStringDeserializer.java]: Yes we need this even more now.  The Kafka Deserializer (including StringDeserializer) is not java serializable.  So if we don't do this on a real storm cluster we will get exceptions when we try to write out the spout.

I can look into trying to support some kind of generics like 
```
public <NK> Builder<NK,V> setKey(Class<? extends Deserializer<NK>> keyDeserializer) {
```
But I really don't know if that works.  I'll try it out and let you know.

**Comment by revans2** [comment]
@harshach and @ppoulosk I just pushed fixes for your review comments.  @harshach you were right I could and did remove StormStringDeserializer.  I think this made the code much better.

**Comment by ppoulosk** [comment]
Thanks, @revans2.  Still +1, NB.

**Comment by harshach** [comment]
+1. @revans2 can you squash the commits.

**Comment by revans2** [comment]
@harshach I am happy to once I get a +1 on #1808 too.  It has almost identical code, but is for master.

**Comment by revans2** [comment]
This branch is also in sync with #1808 now tabs are removed and code is squashed.

**Comment by hmcl** [comment]
+1

**Comment by HeartSaVioR** [comment]
@revans2 
Seems like the commit for JDK 7 is removed so Travis CI shows compile error on JDK 7. Could you address this?

**Comment by revans2** [comment]
It was two jdk8 annotations that I didn't remove as part of rework/upmerging.  I'll get them

**Comment by revans2** [comment]
@HeartSaVioR I removed the unneeded annotation and rebuilt with JDK7

**Comment by HeartSaVioR** [comment]
+1

**Pull request merged**